iotservice/client.go

```diff
@@ -151,18 +151,16 @@ func (c *Client) connectToEventHub(ctx context.Context) (*amqp.Client, string, e
 		return nil, "", err
 	}
 
-	addr := "amqps://" + c.creds.HostName
-	conn, err := amqp.Dial(addr, amqp.ConnSASLPlain(user, pass))
-	if err != nil {
-		return nil, "", err
+	addrDevice := "amqps://" + c.creds.HostName
+	connDevice, errDevice := amqp.Dial(addrDevice, amqp.ConnSASLPlain(user, pass))
+	if errDevice != nil {
+		return nil, "", errDevice
 	}
 	defer func() {
-		if err != nil {
```

Comment:
Nice catch, thanks!
Could you just make it one-line change `defer conn.Close()` without providing a closure it equals to:
```
defer func(conn *Client) {
	conn.Close()
}(conn)
```
so there's be no need to introduce new variables.

Reply:
I think, Your snippet is beautiful!
But I feel like the maintainability is low...

I want to divide, But I respect Owner idea.
Tell me what you think.

Reply:
I'd like a very small commit that solve a particular problem, since even clear refactoring takes time to test the whole codebase for regressions

Reply:
ok! I deleted new variables! https://github.com/goautomotive/iothub/pull/3/commits/cdfba83080ed53f65085d94646e89c0e208b5d73